persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/models/ModelEntity.java

```diff
@@ -178,8 +200,13 @@ public Map<String, Object> toMap() {
     map.put("purge_timestamp", this.getPurgeTimestamp());
     map.put("to_purge_timestamp", this.getToPurgeTimestamp());
     map.put("last_update_timestamp", this.getLastUpdateTimestamp());
-    map.put("properties", this.getProperties());
-    map.put("internal_properties", this.getInternalProperties());
+    if (databaseType.equals(DatabaseType.POSTGRES)) {
+      map.put("properties", toJsonbPGobject(this.getProperties()));
```

Comment:
suggestion: use an interface for converting values to driver-specific types and provide different implementations per `DatabaseType` (it could even be obtained from `DatabaseType.converter()` perhaps) instead of `if`s.